craft_parts/sources/snap_source.py

```diff
@@ -96,33 +96,39 @@ def provision(
         if src:
             snap_file = src
         else:
-            snap_file = os.path.join(self.part_src_dir, os.path.basename(self.source))
-        snap_file = os.path.realpath(snap_file)
+            snap_file = self.part_src_dir / os.path.basename(self.source)
+        snap_file = snap_file.resolve()
 
         if clean_target:
             with tempfile.NamedTemporaryFile() as tmp_file:
                 tmp_snap = tmp_file.name
-                shutil.move(snap_file, tmp_snap)
+                shutil.move(str(snap_file), tmp_snap)  # path-like arg requires 3.9
                 shutil.rmtree(dst)
                 os.makedirs(dst)
-                shutil.move(tmp_snap, snap_file)
-
-        snap_dir, snap = os.path.split(snap_file)
+                shutil.move(tmp_snap, str(snap_file))
 
         # unsquashfs [options] filesystem [directories or files to extract]
         # options:
         # -force: if file already exists then overwrite
         # -dest <pathname>: unsquash to <pathname>
-        with tempfile.TemporaryDirectory(prefix=snap_dir) as temp_dir:
-            extract_command = ["unsquashfs", "-force", "-dest", temp_dir, snap_file]
+        with tempfile.TemporaryDirectory(prefix=str(snap_file.parent)) as temp_dir:
+            extract_command = [
+                "unsquashfs",
+                "-force",
+                "-dest",
+                temp_dir,
+                str(snap_file),
```

Comment:
this goes to subprocess too

Reply:
In this case the list goes to `_run_output` which expects `List[str]`.

Reply:
Hmm, it seems that it will require more changes, so I'll leave this as is and fix it later in a different os.path-to-pathlib refactoring round.

Reply:
Changed `_run_output` argument type to `Sequence` and fixed a couple of tests so that arguments can be passed as subprocess expects them.